### DISCUSSION TITLE
Initial load or regular full load based on connections

### DISCUSSION THREAD

**Discussion by thfries** [eclipse-ditto/ditto#1526] [Ideas]
Ditto connections allow to send updates of twin data to consumer applications that are interested. But in order to start, a consumer must be initialized with all data as a starting point. A similar process might be needed on a regular basis to align consumers to the source state, if they are only delivered with eventual consistency.

I wonder if there is a good approach for that.

An initial load can be done manually by polling all data by a Thing search, as this is a one time execution. But especially for the periodic execution a build in solution via the existing channel of the connection would be good. E.g. force Ditto to push a full search result to a connection?

Is there an idea around or a solution?

**Comment by thjaeckle** [comment]
Did you consider to use the [search protocol](https://www.eclipse.org/ditto/protocol-specification-things-search.html) via a connection and periodically trigger a "full search", streaming back all results?
The Ditto Java Client already provides a Java API for that facade (via WebSocket) as well - if that would be an option.
